sdk/communication/azure-communication-chat/azure/communication/chat/_shared/user_credential.py

```diff
@@ -3,84 +3,139 @@
 # Licensed under the MIT License. See License.txt in the project root for
 # license information.
 # --------------------------------------------------------------------------
-from threading import Lock, Condition
-from datetime import timedelta
-from typing import ( # pylint: disable=unused-import
-    cast,
-    Tuple,
-)
 
+from threading import Lock, Condition, Timer, TIMEOUT_MAX, Event
+from datetime import timedelta
+from typing import Any
+import six
 from .utils import get_current_utc_as_int
-from .user_token_refresh_options import CommunicationTokenRefreshOptions
+from .utils import create_access_token
 
 
 class CommunicationTokenCredential(object):
     """Credential type used for authenticating to an Azure Communication service.
-    :param str token: The token used to authenticate to an Azure Communication service
-    :keyword token_refresher: The token refresher to provide capacity to fetch fresh token
-    :raises: TypeError
+    :param str token: The token used to authenticate to an Azure Communication service.
+    :keyword token_refresher: The sync token refresher to provide capacity to fetch a fresh token.
+     The returned token must be valid (expiration date must be in the future).
+    :paramtype token_refresher: Callable[[], AccessToken]
+    :keyword bool proactive_refresh: Whether to refresh the token proactively or not.
+     If the proactive refreshing is enabled ('proactive_refresh' is true), the credential will use
+     a background thread to attempt to refresh the token within 10 minutes before the cached token expires,
+     the proactive refresh will request a new token by calling the 'token_refresher' callback.
```

Comment:
Could we add a note to the documentation that if `proactive_refresh` is enabled, the Credential object must be either run within a context manager, or `close` called once the object has been finished with.